sql/src/main/java/org/apache/druid/sql/calcite/planner/SqlParameterizerShuttle.java

```diff
@@ -129,16 +129,17 @@ private SqlNode createArrayLiteral(Object value, int posn)
     List<SqlNode> args = new ArrayList<>(list.size());
     for (int i = 0, listSize = list.size(); i < listSize; i++) {
       Object element = list.get(i);
-      if (element == null) {
-        throw InvalidSqlInput.exception("parameter [%d] is an array, with an illegal null at index [%d]", posn + 1, i);
-      }
       SqlNode node;
-      if (element instanceof String) {
+      if (element == null) {
+        node = SqlLiteral.createNull(SqlParserPos.ZERO);
+      } else if (element instanceof String) {
         node = SqlLiteral.createCharString((String) element, SqlParserPos.ZERO);
       } else if (element instanceof Integer || element instanceof Long) {
         // No direct way to create a literal from an Integer or Long, have
         // to parse a string, sadly.
         node = SqlLiteral.createExactNumeric(element.toString(), SqlParserPos.ZERO);
+      } else if (element instanceof Double || element instanceof Float) {
+        node = SqlLiteral.createApproxNumeric(element.toString(), SqlParserPos.ZERO);
       } else if (element instanceof Boolean) {
         node = SqlLiteral.createBoolean((Boolean) value, SqlParserPos.ZERO);
```

Comment:
:bomb: if is for `element` cast is for `value`  :headstone: 

please also add a test for:
```
                    new SqlParameter(SqlType.ARRAY, Arrays.asList(true,false))
```

Reply:
good catch. corrected it now.